include/flamegpu/runtime/environment/DeviceEnvironmentDirectedGraph.cuh

```diff
@@ -0,0 +1,653 @@
+             * @note Edge indexes are not constant, they can change if edges are updated causing the graph's CSR to be rebuilt
+             */
+            __device__ unsigned int getIndex() const { return edge_index; }
+            /**
+             * Returns the value for the current edge attached to the named property
+             * @param property_name Name of the property
+             * @tparam T type of the property
+             * @tparam N Length of property name (this should be implicit if a string literal is passed to property_name)
+             * @return The specified variable, else 0x0 if an error occurs
+             */
+            template<typename T, flamegpu::size_type N>
+            __device__ T getProperty(const char(&property_name)[N]) const;
+            /**
+             * Returns the specified property array element from the current edge attached to the named property
+             * @param property_name name used for accessing the property, this value should be a string literal e.g. "foobar"
+             * @param index Index of the element within the property array to return
+             * @tparam T Type of the edge property being accessed
+             * @tparam N The length of the edge's array property, as set within the model description hierarchy
+             * @tparam M Length of property name (this should be implicit if a string literal is passed to property_name)
+             * @throws exception::DeviceError If name is not a valid property within the edge (flamegpu must be built with SEATBELTS enabled for device error checking)
+             * @throws exception::DeviceError If T is not the type of property 'name' within the message (flamegpu must be built with SEATBELTS enabled for device error checking)
+             * @throws exception::DeviceError If index is out of bounds for the property array specified by name (flamegpu must be built with SEATBELTS enabled for device error checking)
+             */
+            template<typename T, flamegpu::size_type N, unsigned int M>
+            __device__ T getProperty(const char(&property_name)[M], const unsigned int &index) const;
+            /**
+             * Returns the source vertex indexof this edge
+             * @note getEdgeDestination() is not provided, as this value is required to access an edge via iteration so should already be available
+             */
+            __device__ unsigned int getEdgeSource() const;
+        };
+        /**
+         * Stock iterator for iterating DeviceEnvironmentDirectedGraph::Edge objects
+         */
+        class iterator {
+            /**
+             * The edge returned to the user
+             */
+            Edge _edge;
+
+         public:
+            /**
+             * Constructor
+             * This iterator is constructed by DeviceEnvironmentDirectedGraph::InEdgeFilter::begin()(id_t)
+             * @see DeviceEnvironmentDirectedGraph::edgesJoining()(id_t)
```

Comment:
Same applies for edges joining as edges leaving, a few occurences in comments in dif files that I may not mark all of, so worth a search. 

```suggestion
             * @see DeviceEnvironmentDirectedGraph::inEdges()(id_t)
```